### DISCUSSION TITLE
Is there a way to keep the text input (or any form control) visible?

### DISCUSSION THREAD

**Discussion by pankajparashar** [charmbracelet/huh#488] [Q&A] [answered]
If I add a text input to the code, input a value, and press enter, the input disappears. Is there a way to prevent form controls from disappearing?

```
huh.NewInput().
    Title("What’s for lunch?").
    Prompt("?").
    Validate(isFood).
    Value(&lunch).
    Run()
```

**Comment by jonas-grgt** [comment, marked as answer]
Out of the box this is not supported but, I have submitted a PR that would make this possible, #469 
Unfortunately it is still waiting for feedback.

**Comment by pankajparashar** [comment]
@jonas-grgt This is great work! Can you elaborate on how the developers can use it to prevent the form from disappearing? Is there a doc I can refer to?

**Reply by jonas-grgt** [reply]
@pankajparashar thanks

There is a public property you can set on `Form` named `QuitAfterSubmit` it defaults to false to make the behavior backwards compatible. Simply setting it to true will allow for not letting the form quit or disappear after submitting.

https://github.com/charmbracelet/huh/blob/33640667617ef83587c43a720ebac3e8032ec9ef/form.go#L79

I was thinking of changing the property to an explicit setter see #469 for more details. Any feedback more than welcome.

**Reply by pankajparashar** [reply]
If you check out my code sample in the Question, I'm not using a `Form`. Just using `huh.NewInput()`, Is it possible to make it work with it? I'm thinking of something like the following,

```
huh.NewInput().
    Title("What’s for lunch?").
    Prompt("?").
    Validate(isFood).
    Value(&lunch).
    WithQuitAfterSubmit(false).
    Run()
```

**Reply by jonas-grgt** [reply]
Sorry I missed that you were using an isolated field.

In theory it could work but not with the built in `Input.Run` nor `form.Run` methods.

You could create your own field `Run` method that makes use of `QuitAfterSubmit` but then you also need to "override" the `form.Run` method.

Here's some demo code I put together

```go
func main() {
	field := huh.NewInput().
		Title("What’s for lunch?").
		Prompt("?")
	Run(field)
}

func Run(field huh.Field) error {
	group := huh.NewGroup(field)
	form := huh.NewForm(group).WithShowHelp(false)
	form.QuitAfterSubmit = false
	return form.Run()
}
```

**Reply by pankajparashar** [reply]
Amazing stuff Jonas. This was very helpful. I'll try out your suggestions. Thanks.
